adafruit_rfm/rfm9x.py

```diff
@@ -521,6 +521,7 @@ def read_fifo(self) -> bytearray:
         """Read the data from the FIFO."""
         # Read the length of the FIFO.
         fifo_length = self.read_u8(_RF95_REG_13_RX_NB_BYTES)
+        packet = None  # return None if FIFO empty
```

Comment:
If this can return `None`, then the type annotation should specify `Optional[bytearray]`.  Is it better to return an empty bytearray?  Not entirely sure what the utility of each case is and which would be better.

Reply:
This might be important since the function that calls read_fifo checks len(packet).  Please confirm.  But this will probably not work on None.  In that case bytearray(0) is probably the correct solution.  I will change my test code to set packet = bytearray(0).  Unless I report back assume it's working that way.

Reply:
And we're still left with the mystery about how read_fifo could be called with nothing in the FIFO - which was happening to me the other day.  I can't tell if that's happening anymore with my rig.  Nonetheless, I think this change and those in the other two files is best practice.

Reply:
> If this can return `None`, then the type annotation should specify `Optional[bytearray]`. Is it better to return an empty bytearray? Not entirely sure what the utility of each case is and which would be better.

Good catch... I want to think about this a bit -- the calling code  has a confusing check:
```
                    if len(packet) < 5:
                        # reject the packet if it is too small to contain the RadioHead Header
                        packet = None
```
which will clearly now fail if read_fifo returns None.....  I think it best to use your original suggestion to make the  return type annotation Optional[bytearray] and refactor the code to always check for `packet is not None`   before checking len(packet). This way read_fifo will be consistent with receive in returning the Optional[bytearray]. Does that make sense?

Reply:
I changed it to Optional[bytearray] in the 3 files and in rfm_common.py  changed code to avoid testing let(packet) id packet is None.  Let me know what you think.